src/search/landmarks/landmark_factory.h

```diff
@@ -38,70 +33,53 @@ class LandmarkFactory {
     virtual bool supports_conditional_effects() const = 0;
 
 protected:
+    explicit LandmarkFactory(const options::Options &opts);
+
     std::shared_ptr<LandmarkGraph> lm_graph;
-    AbstractTask *lm_graph_task;
+    const bool reasonable_orders;
+    const bool only_causal_landmarks;
+    const bool disjunctive_landmarks;
+    const bool conjunctive_landmarks;
+    const bool no_orders;
 
+    // TODO: Directly use no_orders in LandmarkFactoryHM and remove this
     bool use_orders() const {return !no_orders;}   // only needed by HMLandmark
 
-    virtual void generate_landmarks(const std::shared_ptr<AbstractTask> &task, Exploration &exploration) = 0;
-    void generate(const TaskProxy &task_proxy, Exploration &exploration);
-    void discard_noncausal_landmarks(const TaskProxy &task_proxy, Exploration &exploration);
```

Comment:
I think we should document (i.e., add a TODO) that this method (and a few others) are now duplicated in HMLandmarks and RelaxationLandmarks.